securedrop_export/main.py

```diff
@@ -1,46 +1,180 @@
+import os
+import shutil
+import platform
 import logging
+import sys
 
-from securedrop_export import export
-from securedrop_export.exceptions import ExportStatus
-from securedrop_export.print.actions import (
-    PrintExportAction,
-    PrintTestPageAction,
-    PrintPreflightAction,
-)
-from securedrop_export.disk.actions import (
-    DiskTestAction,
-    DiskExportAction,
-    USBTestAction,
-)
+from securedrop_export.archive import Archive, Metadata
+from securedrop_export.command import Command
+from securedrop_export.status import BaseStatus
+from securedrop_export.directory_util import safe_mkdir
+from securedrop_export.exceptions import ExportException
+
+from securedrop_export.disk.service import Service as ExportService
+from securedrop_export.print.service import Service as PrintService
+
+from logging.handlers import TimedRotatingFileHandler, SysLogHandler
+from securedrop_export import __version__
+
+DEFAULT_HOME = os.path.join(os.path.expanduser("~"), ".securedrop_export")
+LOG_DIR_NAME = "logs"
+EXPORT_LOG_FILENAME = "export.log"
 
 logger = logging.getLogger(__name__)
 
 
-def __main__(submission):
-    submission.extract_tarball()
+class Status(BaseStatus):
+    """
+    Status values that can occur during initialization.
+    """
+
+    ERROR_LOGGING = "ERROR_LOGGING"
+    ERROR_GENERIC = "ERROR_GENERIC"
+    ERROR_FILE_NOT_FOUND = "ERROR_FILE_NOT_FOUND"
+
+
+def entrypoint():
+    """
+    Entrypoint method (Note: a method is required for setuptools).
+    Configure logging, extract tarball, and run desired export service,
+    exiting with return code 0.
```

Comment:
Copying the explanation form below.

```suggestion
    Entrypoint method (Note: a method is required for setuptools).
    Configure logging, extract tarball, and run desired export service,
    exiting with return code 0.
    
    Non-zero exit values will cause the system to try alternative
    solutions for mimetype handling, which we want to avoid.
```

I would say this everywhere we mention returning with code 0, because it seems pretty important.